crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -1887,6 +1892,51 @@ mod tests {
         );
     }
 
+    #[async_test]
+    async fn test_read_receipt_can_trigger_a_notable_update_reason() {
+        // Given a logged-in client
+        let client = logged_in_base_client(None).await;
+        let mut room_info_notable_update_stream = client.room_info_notable_update_receiver();
+
+        // When I send sliding sync response containing a new room.
+        let room_id = room_id!("!r:e.uk");
+        let room = v4::SlidingSyncRoom::new();
+        let response = response_with_room(room_id, room);
+        client.process_sliding_sync(&response, &()).await.expect("Failed to process sync");
+
+        // Then a room info notable update is NOT received.
+        assert_matches!(
+            room_info_notable_update_stream.recv().await,
+            Ok(RoomInfoNotableUpdate { room_id: received_room_id, reasons: received_reasons }) => {
+                assert_eq!(received_room_id, room_id);
+                dbg!(&received_reasons);
+                assert!(!received_reasons.contains(RoomInfoNotableUpdateReasons::READ_RECEIPT));
+            }
+        );
+
+        // When I send sliding sync response containing a couple of events with no read receipt.
+        let room_id = room_id!("!r:e.uk");
+        let events = vec![
+            make_raw_event("m.room.message", "$3"),
+            make_raw_event("m.room.message", "$4"),
+            make_raw_event("m.read", "$5"),
```

Comment:
Technically this is wrong, as this will create a `m.read` event with the content of a `m.room.message` event; it seems `compute_unread_counts()` still does something with it (store it, maybe?), but this could be a bit better…

Reply:
There is no `RoomReadReceipt` before, and now there is one, so the previous and the new one are different, which triggers something. Do you have a better test in mind?

Reply:
All I'm saying is that the `m.read` event should be created in a better way (with a proper `content` property), because it doesn't make sense now :)

Reply:
Since you've approved the PR, I understand it's fine for this particular test. The read receipt mechanism is already pretty well test. Here we only want to test the notable update reason is triggered, which is fine for the moment.